pywbem_mock/_instancewriteprovider.py

```diff
@@ -277,114 +220,117 @@ def __init__(self, cimrepository=None):
-            The name of the CIM namespace in the CIM repository (case
-            insensitive). Must not be `None`. Leading or trailing slash
-            characters are ignored.
+            The name of the CIM namespace in which the CIM instance is to be
+            created, in any lexical case, and with leading and trailing slash
+            characters removed.
 
           NewInstance (:class:`~pywbem.CIMInstance`):
             A representation of the CIM instance to be created.
 
-            The ``classname`` attribute of this object specifies the creation
-            class for the new instance.
+            This object is a deep copy of the original client parameter, and may
+            be modified by the provider as needed, before storing it in the
+            CIM repository.
 
-            The namespace of the ``path`` attribute must be `None` or the
-            same as the namespace parameter.
+            The ``classname`` attribute of this object will specify the
+            creation class for the new instance, in any lexical case.
 
-            The `properties` attribute of this object specifies initial
-            property values for the new CIM instance.
+            The ``properties`` attribute of this object will specify the
+            initial property values for the new CIM instance, with property
+            names in any lexical case. Key properties may or may not be
+            included.
 
-            Instance-level qualifiers have been deprecated in CIM, so any
-            qualifier values specified using the `qualifiers` attribute
-            of this object are ignored.
+            The ``path`` attribute of this object will be `None`.
+
+            The ``qualifiers`` attribute of this object, if non-empty, should
+            be ignored by the provider, because instance-level qualifiers have
+            been deprecated in CIM.
 
         Returns:
 
             :class:`~pywbem.CIMInstanceName`: Instance path of the new CIM
             instance.
 
         Raises:
-
-            :exc:`~pywbem.CIMError`: (CIM_ERR_INVALID_NAMESPACE)
-            :exc:`~pywbem.CIMError`: (CIM_ERR_INVALID_PARAMETER)
-            :exc:`~pywbem.CIMError`: (CIM_ERR_INVALID_CLASS)
-            :exc:`~pywbem.CIMError`: (CIM_ERR_ALREADY_EXISTS)
+            :exc:`~pywbem.CIMError`: The provider may raise any CIMError, and
+              typically raises:
+              - CIM_ERR_INVALID_PARAMETER
+              - CIM_ERR_ALREADY_EXISTS
         """
 
-        # Parameter types are already checked by WBEMConnection operation
-        assert isinstance(namespace, six.string_types)
-        assert isinstance(NewInstance, CIMInstance)
-
-        new_instance = NewInstance.copy()
+        # Get the creation class with all exposed properties and qualifiers.
+        # Since the existence of the class has already been verified, this
+        # will always succeed.
+        creation_class = self.get_class(
```

Comment:
Again, maybe bare_class since this is readonly

Reply:
We need all inherited properties.

Reply:
DONE - It turns out I was not aware that each class in the repository is stored with all exposed (defined + inherited, with overrides resolved) properties (and methods, fwiw). So we can go to the class repository and simply get it. Changed the code in both the dispatcher and the provider accordingly.